client/driver/executor/executor_test.go

```diff
@@ -149,6 +149,50 @@ func TestExecutor_IsolationAndConstraints(t *testing.T) {
 	}
 }
 
+func TestExecutor_DestroyCgroup(t *testing.T) {
+	testutil.ExecCompatible(t)
+
+	execCmd := ExecCommand{Cmd: "/bin/bash", Args: []string{"-c", "/usr/bin/yes"}}
+	ctx := testExecutorContext(t)
+	ctx.LogConfig.MaxFiles = 1
+	ctx.LogConfig.MaxFileSizeMB = 300
+	defer ctx.AllocDir.Destroy()
+
+	ctx.FSIsolation = true
+	ctx.ResourceLimits = true
+	ctx.UnprivilegedUser = true
+
+	executor := NewExecutor(log.New(os.Stdout, "", log.LstdFlags))
+	ps, err := executor.LaunchCmd(&execCmd, ctx)
+	if err != nil {
+		t.Fatalf("error in launching command: %v", err)
+	}
+	if ps.Pid == 0 {
+		t.Fatalf("expected process to start and have non zero pid")
+	}
+	go func() {
```

Comment:
So the Wait is going to block, and we won't be able to call Exit on the executor.